sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/CommunicationIdentityClientBuilder.java

```diff
@@ -203,6 +223,22 @@ private CommunicationIdentityClientImpl createServiceImpl() {
         return clientBuilder.buildClient();
     }
 
+    private HttpPipelinePolicy createHttpPipelineAuthPolicy() {
+        if (this.tokenCredential != null && this.accessKeyCredential != null) {
+            throw logger.logExceptionAsError(
+                new IllegalArgumentException("Both 'credential' and 'accessKey' are set. Just one may be used."));
+        }
```

Comment:
This is good! Since we have this check here, we don't need to do any validation in the setter methods above.